var/spack/repos/builtin/packages/libpng/package.py

```diff
@@ -35,14 +35,25 @@ class Libpng(AutotoolsPackage):
 
     depends_on('zlib@1.0.4:')  # 1.2.5 or later recommended
 
+    #
+    # Only set this if using a more recent GCC and building for ARM.
+    #
+    variant('arm-neon', default=False, description='Build ARM neon support')
```

Comment:
@hppritcha do you know what GCC version this requires? You could add a `conflicts` statement to make sure users don't try with a version of GCC that is too old.

(Ideally you could also do that to restrict the variant to ARM but that's difficult without support for syntax like `!platform=arm` which isn't supported right now)